### PR TITLE
Restructure project & integrate conan + doctest

### PR DESCRIPTION
**Pull request by shrijitsingh99**
Restructure project
Integrate conan and doctest

If everything is fine here, I will give the go-ahead and integrate the with a simple CI.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,24 +1,18 @@
...
-    add_library(mmul_cuda STATIC src/mmul.cu)
-endif()
+add_executable(main_test test/main.cpp)
+target_link_libraries(main_test doctest::doctest executor)
 
-add_executable(executor_example src/executor.cpp)
-
-target_link_libraries(executor_example unified_api_executor mmul_cuda)
-set_target_properties(executor_example PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
+doctest_discover_tests(main_test)
```
Comment (kunaltyagi) [inline CMakeLists.txt:18]: You can provide args for better error reporting such as:
```cmake
doctest_discover_tests(
  constexpr_tests
  TEST_PREFIX
  "constexpr."
  EXTRA_ARGS
  -s
  --reporter=xml
  --out=constexpr.xml)
```

File: application/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.8)
+
+project(application VERSION 0.0.1 LANGUAGES)
```
Comment (kunaltyagi) [inline application/CMakeLists.txt]: Something after languages? Is `languages` necessary?

File: application/test/mmul.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
+//
+// Created by shrijit on 6/9/20.
+//
+
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
```
Comment (kunaltyagi) [inline application/test/mmul.cpp:5]: I'd recommend creating a header for doctest and including that instead. All you config and macros for testing can go in that

**Review by kunaltyagi — COMMENTED** [review]
Just a quick view over

File: application/test/mmul.cpp (hunk 3 above)
Comment (shrijitsingh99) [reply application/test/mmul.cpp:5]: Sounds good. I will need to make a common module for that where I can place all the common stuff. Will add this to my TODO.

File: application/CMakeLists.txt (hunk 2 above)
Comment (shrijitsingh99) [reply application/CMakeLists.txt]: It's optional for C and CPP. Will remove it.

File: CMakeLists.txt (hunk 1 above)
Comment (shrijitsingh99) [reply CMakeLists.txt:18]: Will add them as needed.

**Comment by shrijitsingh99** [comment]
If there are no further changes will go ahead and merge this in.

File: application/src/mmul.cu [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#include "mmul.cuh"
+#include "../include/mmul.cuh"
```
Comment (kunaltyagi) [inline application/src/mmul.cu]: Why? `target_include_directory` will sort this out

**Review by kunaltyagi — APPROVED** [review]

File: application/src/mmul.cu (hunk 4 above)
Comment (shrijitsingh99) [reply application/src/mmul.cu]: Yea, I guess the IDE did it automatically when I restructured everything.

**Pull request merged**